expertsystem/state/conservation_rules.py

```diff
@@ -232,106 +78,93 @@
+            f"""Check for `.{quantum_number.__name__}` conservation."""
         )
-        rule_class = rule_conditions(
-            variable_conditions=[(quantum_number, [DefinedForAllEdges()])]
-        )(rule_class)
         return rule_class
 
     return decorator
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.Charge)
+@additive_quantum_number_rule(EdgeQuantumNumbers.charge)
 class ChargeConservation(Rule):
     pass
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.BaryonNumber)
+@additive_quantum_number_rule(EdgeQuantumNumbers.baryon_number)
 class BaryonNumberConservation(Rule):
     pass
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.ElectronLN)
+@additive_quantum_number_rule(EdgeQuantumNumbers.electron_lepton_number)
 class ElectronLNConservation(Rule):
     pass
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.MuonLN)
+@additive_quantum_number_rule(EdgeQuantumNumbers.muon_lepton_number)
 class MuonLNConservation(Rule):
     pass
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.TauLN)
+@additive_quantum_number_rule(EdgeQuantumNumbers.tau_lepton_number)
 class TauLNConservation(Rule):
     pass
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.Strangeness)
+@additive_quantum_number_rule(EdgeQuantumNumbers.strangeness)
 class StrangenessConservation(Rule):
     pass
 
 
-@additive_quantum_number_rule(StateQuantumNumberNames.Charmness)
+@additive_quantum_number_rule(EdgeQuantumNumbers.charmness)
 class CharmConservation(Rule):
     pass
 
 
-@rule_conditions(
-    variable_conditions=[
-        (StateQuantumNumberNames.Parity, [DefinedForAllEdges()]),
-        (InteractionQuantumNumberNames.L, [DefinedForInteractionNode()]),
-    ]
-)
 class ParityConservation(Rule):
-    def __call__(self, ingoing_part_qns, outgoing_part_qns, interaction_qns):
+    def __call__(
+        self,
+        ingoing_edge_qns: List[EdgeQuantumNumbers.parity],
+        outgoing_edge_qns: List[EdgeQuantumNumbers.parity],
+        l_mag: NodeQuantumNumbers.l_magnitude,
+    ) -> bool:
         r"""Implement :math:`P_{in} = P_{out} \cdot (-1)^L`."""
-        # is this valid for two outgoing particles only?
-        parity_label = StateQuantumNumberNames.Parity
-        parity_in = reduce(
-            lambda x, y: x * y[parity_label], ingoing_part_qns, 1
-        )
-        parity_out = reduce(
-            lambda x, y: x * y[parity_label], outgoing_part_qns, 1
-        )
-        ang_mom = interaction_qns[InteractionQuantumNumberNames.L].magnitude
-        if parity_in == (parity_out * (-1) ** ang_mom):
-            return True
-        return False
+        if len(ingoing_edge_qns) == 1 and len(outgoing_edge_qns) == 2:
+            parity_in = reduce(lambda x, y: x * y.value, ingoing_edge_qns, 1,)
+            parity_out = reduce(lambda x, y: x * y.value, outgoing_edge_qns, 1)
+            return parity_in == (parity_out * (-1) ** l_mag)
+        return True
```

Comment:
Hmm this is more of a technical trick to make things working. In principle it would be best to define a third state for Parity which would be Unknown (meaning something like Any). Same counts for g and c parity

Reply:
https://github.com/ComPWA/expertsystem/issues/256